components/rule-mgt/org.wso2.carbon.identity.rule.management/src/main/java/org/wso2/carbon/identity/rule/management/dao/impl/RuleManagementDAOImpl.java

```diff
@@ -0,0 +1,282 @@
+                    .setActive(isActive[0])
+                    .build();
+        } catch (DataAccessException e) {
+            throw new RuleManagementServerException("Error while retrieving the rule from the system.", e);
+        }
+    }
+
+    /**
+     * This method will activate the Rule in the database.
+     *
+     * @param ruleId   Rule ID
+     * @param tenantId Tenant ID
+     * @throws RuleManagementException If an error occurs while activating the rule in the database.
+     */
+    @Override
+    public void activateRule(String ruleId, int tenantId) throws RuleManagementException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.executeUpdate(RuleSQLConstants.Query.CHANGE_RULE_STATUS,
+                    statement -> {
+                        statement.setBoolean(RuleSQLConstants.Column.IS_ACTIVE, true);
+                        statement.setString(RuleSQLConstants.Column.UUID, ruleId);
+                        statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                    });
+        } catch (DataAccessException e) {
+            throw new RuleManagementServerException("Error while deactivating the rule in the system.", e);
+        }
+    }
+
+    /**
+     * This method will deactivate the Rule in the database.
+     *
+     * @param ruleId   Rule ID
+     * @param tenantId Tenant ID
+     * @throws RuleManagementException If an error occurs while deactivating the rule in the database.
+     */
+    @Override
+    public void deactivateRule(String ruleId, int tenantId) throws RuleManagementException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.executeUpdate(RuleSQLConstants.Query.CHANGE_RULE_STATUS,
+                    statement -> {
+                        statement.setBoolean(RuleSQLConstants.Column.IS_ACTIVE, false);
+                        statement.setString(RuleSQLConstants.Column.UUID, ruleId);
+                        statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                    });
+        } catch (DataAccessException e) {
+            throw new RuleManagementServerException("Error while deactivating the rule in the system.", e);
+        }
+    }
+
+    private void addRuleToDB(Rule rule, int tenantId) throws TransactionException, RuleManagementServerException {
+
+        InputStream ruleJson = convertRuleToJson(rule);
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        jdbcTemplate.withTransaction(template -> {
+            template.executeInsert(RuleSQLConstants.Query.ADD_RULE,
+                    statement -> {
+                        statement.setString(RuleSQLConstants.Column.UUID, rule.getId());
+                        statement.setBlob(RuleSQLConstants.Column.RULE_INDEX, ruleJson);
```

Comment:
Let's use setBinaryStream(String name, InputStream inputStream, long length) method with the column name reference.
https://github.com/wso2/carbon-utils/blob/90b5d4300fafc5d4abaaadb7e41a9a340429f390/components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/NamedPreparedStatement.java#L226